src/openvic-simulation/economy/Producer.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class Producer {
+	public:
+		ProductionType const& PROPERTY(production_type);
+	};
+
+	class Manufacturer : public virtual Producer {
+	public:
+		Good::good_map_t PROPERTY(stockpile);
+	};
+
+	class ArtisanalProducer final : public Manufacturer {
+	public:
+		fixed_point_t PROPERTY(current_production);
+		Good::good_map_t PROPERTY(current_needs);
+	};
+
+	class NonArtisanalProducer : public virtual Producer {
+	public:
+		fixed_point_t PROPERTY(revenue_yesterday);
+		fixed_point_t PROPERTY(output_quantity_yesterday);
+		fixed_point_t PROPERTY(unsold_quantity_yesterday);
+		fixed_point_t PROPERTY(size_multiplier);
+		ordered_map<Pop const*, Pop::pop_size_t> PROPERTY(employees);
```

Comment:
```suggestion
		ordered_map<Pop*, Pop::pop_size_t> PROPERTY(employees);
```
The `Pop` pointers here should probably be non-const, so that we can do stuff like give them wages.

Reply:
I made it `Pop* const` so the pointer is constant but the pop is mutable.